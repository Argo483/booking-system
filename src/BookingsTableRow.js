import React from "react";
import { isBetween } from "./IsBetween";
import PropTypes from "prop-types";

export const BookingsTableRow = ({ hourToRender, calendarDay, index }) => {
  let backgroundColor = index % 2 === 0 ? "lightgrey" : "darkgrey";
  let hourHasCurrentBooking = false;
  for (const booking of calendarDay.bookingsOnThisDay) {
    if (isBetween(hourToRender, booking.startTime, booking.endTime)) {
      hourHasCurrentBooking = true;
    }
  }
  let newBookingColor = "initial";
  for (const booking of calendarDay.csvBookingsOnThisDay) {
    if (isBetween(hourToRender, booking.startTime, booking.endTime)) {
      newBookingColor = "green";
      if (booking.isOverlappingBooking) {
        newBookingColor = "red";
      }
    }
  }

  return (
    <tr style={{ backgroundColor }}>
      <td>{hourToRender.format("LT")}</td>
      <td
        data-testid="current-booking"
        style={{
          width: "100px",
          backgroundColor: hourHasCurrentBooking ? "green" : "initial",
        }}
      ></td>
      <td
        style={{
          width: "100px",
          backgroundColor: newBookingColor,
        }}
      ></td>
    </tr>
  );
};

BookingsTableRow.propTypes = {
  hourToRender: PropTypes.any,
  calendarDay: PropTypes.any,
  index: PropTypes.number,
};
