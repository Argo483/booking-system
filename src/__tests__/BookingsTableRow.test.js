import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import { BookingsTableRow } from '../BookingsTableRow';

it('renders correctly', () => {
  const tree = renderer
    .create(<BookingsTableRow hourToRender={moment("20111031", "YYYYMMDD")} calendarDay={{bookingsOnThisDay: [], csvBookingsOnThisDay: []}}></BookingsTableRow>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});