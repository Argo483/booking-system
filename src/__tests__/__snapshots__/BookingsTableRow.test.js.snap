// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  style={
    Object {
      "backgroundColor": "darkgrey",
    }
  }
>
  <td>
    12:00 AM
  </td>
  <td
    style={
      Object {
        "backgroundColor": "initial",
        "width": "100px",
      }
    }
  />
  <td
    style={
      Object {
        "backgroundColor": "initial",
        "width": "100px",
      }
    }
  />
</tr>
`;
